infrastructure/file_metadata_service.go

```diff
@@ -46,6 +45,10 @@ func (ms fileMetadataService) GetPublicKey() (string, error) {
 }
 
 func (ms fileMetadataService) GetInstanceID() (string, error) {
+	if ms.metaDataFilePath == "" {
+		return "", nil
+	}
+
```

Comment:
I am sure that this condition is redundant. If file path is not set, or set to the wrong path, `ReadFile ` is going to return the error ((source)[https://github.com/cloudfoundry/bosh-agent/blob/master/infrastructure/file_metadata_service.go#L51]) and bosh-agent will fail. Failing in this case is expected behaviour.